### ISSUE TITLE
Nemo hides filename for .desktop files with execute permission

### ISSUE THREAD

**Issue by cboxdoerfer** [linuxmint/nemo#1404]
How to reproduce:

1. Create a file called `malware.desktop`

2. Add the following content to it:

```
[Desktop Entry]
Name=CV.pdf
Exec=sh -c 'touch ~/MALWARE_WAS_HERE'
Terminal=false
Icon=x-office-document
Type=Application
Categories=Office
```

3. Make it executable

Nemo displays the file like that:
![screenshot from 2017-01-31 18-13-29](https://cloud.githubusercontent.com/assets/6108388/22475957/057be074-e7e1-11e6-87ec-deda273a49a7.png)

Once the user opens the file the Exec entry is executed without any confirmation. By hiding the filename and therefore also the filename extension users can easily be tricked to execute arbitrary code when someone ships files like that in an archive which preserves execute permissions.

How to fix it:

Maybe by don't hiding the filename for .desktop files at all.

For reference, this bug also applies to other file managers:
https://github.com/lxde/pcmanfm-qt/issues/449
https://github.com/mate-desktop/caja/issues/727
https://bugzilla.gnome.org/show_bug.cgi?id=777991

**Comment by clefebvre** [comment]
I agree with the analysis done by the pcmanfm team on this. Imo, the right solution is to interpret desktop files as executables and to show a confirmation dialog by default.

I'll close the PR #1405. Please feel free to open a new one if you're interested in implementing the mentioned solution.

**Comment by ChrisJAllan** [comment]
Is PR #1407 enough? It doesn't make it always ask for confirmation, but follow the executable file settings on whether to run, edit, or ask.

**Comment by ByteHazard** [comment]
IMO, only on the desktop view should the *.desktop filename be hidden, and then only with some security checks in place (as suggested in related threads).

Hiding the filename for desktop files opens a can of worms, from security issues (Windows has always had a large variety of exploits, based on a similar Windows feature), to the inability to rename the .desktop files (it requires installing an alternative file manager to change the file extension without resorting to command line trickery, for example).

**Comment by Moilleadoir** [comment]
I don’t think this functionality in itself could be considered a bug. A lack of a prompt is a bug, but not merely the name displayed. This is the standard behaviour in Thunar and Nautilus. It also has the benefit of allowing a user to easily find the right .desktop file to edit, drag to a dock, etc. This applies doubly to people not using an English interface. On my system it’s much easier to find a file called **Inrochtaineacht** (“Accessibility”) than `xfce4-accessibility-settings.desktop` or **Monatóir an Chórais** than `gnome-system-monitor.desktop`.

**Comment by smurphos** [comment]
@mtwebster - did https://github.com/mtwebster/nemo/commit/31c4787cf5d80abc988615d855d2c9691e6ae01a ever get into the main codebase?

**Comment by xenopeek** [comment]
Testing in Nemo 4.6.4 with the `malware.desktop` file as in the issue, it doesn't hide the file name and prompts the user it is an untrusted file when launched:

![image](https://user-images.githubusercontent.com/21159972/85917389-12c9bd80-b85a-11ea-8b53-02a027b2552a.png)

**Comment by icarter09** [comment]
Nemo 4.8.4
Mint 20.1 64bit

Still an issue. Looks like that code never made into the main codebase.

**Comment by Jeremy7701** [comment]
Nemo 4.8.4

Provided you don't create the .desktop file as executable, you get a warning if you try to run it:

![Screenshot from 2021-01-31 19-59-28](https://user-images.githubusercontent.com/52585119/106396357-52174880-63ff-11eb-924a-f05363749380.png)

Unfortunately, you can elect to run successfully a non-executable desktop file.

In addition, what is the difference between having a text `.sh` file [marked as executable] and a text `.desktop` file [marked as executable]?
Why does one get no special hidden treatment whilst the other is a hidden file?

**Comment by torndar** [comment]
As a power user I'd also really like to at least have the option of showing the file name in a file browsing window. I've run both Windows and FreeBSD (for servers) for many years and turning off extension hiding and showing hidden files is one of the very first changes I make on any computer I get.

**Comment by Jeremy7701** [comment]
What changes would you make if you had easy access to the  .desktop files?
`locate .desktop` will list them in the CLI - there's rather a lot of them.
Since they are owned by root, you will need admin privileges to change them.

**Comment by GitHubIsToxic** [comment]
I'd like to repeat my previous suggestion (7 years later): hide the .desktop extension only on the actual desktop, but always show it in the file manager.

Wouldn't that make total sense anyway, given the actual file extension?!

**Comment by torndar** [comment]
If nothing else the inconsistency bothers me. If I'm shown the file extension for everything other file in the folder in the browser except this one case it's confusing. I didn't expect it to behave that way and was briefly confused. 

As far as doing something with them I've created them for .AppImage applications and could see myself creating other shortcuts or quick commands with them. I'm new to Linux as a desktop and maybe Nemo isn't the right file manager for me. I switched to it from the default because the default file manager doesn't even let me type a path which is mind boggling to me.

[semi rant]
Getting off topic but maybe Fedora is also too "typical end user focused" and I need to try a different distro altogether? I tried Ubuntu and ran into issues with Snap packages (and other things) not leaving me enough control there. I don't want to have to pick every package to have a useful system which is the impression I get reading about Arch but I also don't want to have my hands tied by patterns to help new users. I don't hate defaults to help new users but having the ability to "unlock" "power uses" has value in my opinion. Heck have a "show advanced settings" setting even to make the common ones easy to find, I just get frustrated by "convenience" or "simplification" (hiding true paths) that gets in my way and can't be turned off.
